src/moonlink/src/table_handler.rs

```diff
@@ -645,15 +645,24 @@ impl TableHandler {
                     }
                     // TODO: Unify error handling pattern with other background events (e.g. DataCompactionResult)
                     TableEvent::IndexMergeResult { index_merge_result } => {
-                        table.record_index_merge_completion(&index_merge_result);
-                        table.set_file_indices_merge_res(index_merge_result);
-                        table_handler_state.mark_index_merge_completed().await;
-                        // Check whether need to drop table.
-                        if table_handler_state.special_table_state == SpecialTableState::DropTable
-                            && table_handler_state.can_drop_table_now(table.has_ongoing_flush())
-                        {
-                            drop_table(&mut table, event_sync_sender).await;
-                            return;
+                        match index_merge_result {
+                            Ok(index_merge_result) => {
+                                table.record_index_merge_completion(&index_merge_result);
+                                table.set_file_indices_merge_res(index_merge_result);
+                                table_handler_state.mark_index_merge_completed().await;
```

Comment:
Please mark index merge as completed even when index merge fails. 

Context:
- We only allow at most one concurrent index merge operation in the background
- Even if this index merge fails, we could still retry in the future.
- If you didn't mark it as completed, we can never retry.

I highly suggest follow the same process pattern as data compaction.